picard/ui/scripteditor.py

```diff
@@ -0,0 +1,642 @@
+        """
+        override = {'file_naming_format': self.get_script()}
+        self.examples.update_examples(override)
+        self.display_examples()
+
+    def display_examples(self, send_signal=True):
+        """Update the display of the before and after file naming examples.  Optionally emits an `update` signal.
+
+        Args:
+            send_signal (bool, optional): Determines if an `update` signal is emitted. Defaults to True.
+        """
+        self.ui.example_filename_before.clear()
+        self.ui.example_filename_after.clear()
+        self.current_row = -1
+
+        examples = self.examples.get_examples()
+        for before, after in sorted(examples, key=lambda x: x[1]):
+            self.ui.example_filename_before.addItem(before)
+            self.ui.example_filename_after.addItem(after)
+
+        if send_signal:
+            self.signal_update.emit()
+
+    def toggle_wordwrap(self):
+        """Toggles wordwrap in the script editing textbox.
+        """
+        if self.ui.file_naming_word_wrap.isChecked():
+            self.wordwrap = QtWidgets.QTextEdit.WidgetWidth
+        else:
+            self.wordwrap = QtWidgets.QTextEdit.NoWrap
+        self.ui.file_naming_format.setLineWrapMode(self.wordwrap)
+
+    def import_script(self):
+        """Import the current script from an external text file.
+        """
+        dialog_title = N_("Import Script File")
+        dialog_file_types = N_("All Files") + " (*);;" + N_("Picard Script Files") + " (*.pts)"
+        options = QtWidgets.QFileDialog.Options()
+        options |= QtWidgets.QFileDialog.DontUseNativeDialog
+        filename, _ = QtWidgets.QFileDialog.getOpenFileName(self, dialog_title, self.default_script_directory, dialog_file_types, options=options)
+        if filename:
+            log.debug('Importing naming script file: %s' % filename)
+            try:
+                with open(filename, 'r', encoding='utf8') as i_file:
+                    self.set_script(i_file.read())
+            except OSError as error:
+                error_message = 'Error importing file "{0}": {1}'.format(filename, error.strerror)
+                log.error(error_message)
+                self.display_error(OptionsCheckError(N_("File Error"), error_message))
+
+    def export_script(self):
+        """Export the current script to an external text file.
+        """
+        script_text = self.get_script()
+        if script_text:
+            default_path = os.path.normpath(os.path.join(self.default_script_directory, self.default_script_filename))
+            dialog_title = N_("Export Script File")
+            dialog_file_types = N_("All Files") + " (*);;" + N_("Picard Script Files") + " (*.pts)"
+            options = QtWidgets.QFileDialog.Options()
+            options |= QtWidgets.QFileDialog.DontUseNativeDialog
+            filename, _ = QtWidgets.QFileDialog.getSaveFileName(self, dialog_title, default_path, dialog_file_types, options=options)
```

Comment:
Don't use `_` for unused variables, as it is defined as builtin `_()` for i18n purposes (gettext).